### PR TITLE
Feature/rework java ci

### PR DESCRIPTION
**Pull request by sebastienrospars**

### CODE REVIEW

File: .github/workflows/java-microservice-ci-default.yml [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+        default: "eu-west-1"
+      ecr_url:
+        description: "The ECR url"
+        type: string
+        required: true
+      chart_museum_url:
+        description: "The ChartMuseum URL"
+        type: string
+        required: true
+      enable_trivy_exit:
```
Comment (alexandre-cartapanis) [inline .github/workflows/java-microservice-ci-default.yml:58]: Pourquoi désactiver ?
Si on désactive ca, faut aussi pouvoir désactiver les lints, la sonar gate, ...

File: .github/workflows/java-microservice-ci-default.yml [hunk 2]
```diff
@@ -0,0 +1,679 @@
...
+        required: true
+      sonar_organization:
+        description: "The sonar organization"
+        type: string
+        required: true
+      sonar_host_url:
+        description: "The sonar host url"
+        type: string
+        required: true
+      sonar_project_key:
```
Comment (alexandre-cartapanis) [inline .github/workflows/java-microservice-ci-default.yml:37]: always org + project_name no ?

File: .github/workflows/java-microservice-ci-default.yml [hunk 3]
```diff
@@ -0,0 +1,679 @@
...
+        description: "The java version (default to 17)"
+        type: string
+        required: false
+        default: "17"
+      helm_version:
+        description: "The helm version (default to 3.12.0)"
+        type: string
+        required: false
+        default: "3.12.0"
+      slack_channel_id_ci:
```
Comment (alexandre-cartapanis) [inline .github/workflows/java-microservice-ci-default.yml:21]: Default to C03KTE2LJQY

File: .github/workflows/java-microservice-ci-default.yml [hunk 4]
```diff
@@ -0,0 +1,679 @@
...
+        description: "The CI Slack channel's id"
+        type: string
+        required: true
+      slack_channel_id_ci_release:
```
Comment (alexandre-cartapanis) [inline .github/workflows/java-microservice-ci-default.yml:25]: default to C04LQJQNCP9

File: .github/workflows/java-microservice-ci-default.yml [hunk 5]
```diff
@@ -0,0 +1,679 @@
...
+        description: "The CI release Slack channel's id"
+        type: string
+        required: true
+      sonar_organization:
```
Comment (alexandre-cartapanis) [inline .github/workflows/java-microservice-ci-default.yml:29]: default to bruxless

File: .github/workflows/java-microservice-ci-default.yml [hunk 6]
```diff
@@ -0,0 +1,679 @@
...
+        description: "The sonar organization"
+        type: string
+        required: true
+      sonar_host_url:
```
Comment (alexandre-cartapanis) [inline .github/workflows/java-microservice-ci-default.yml:33]: default to https://sonarcloud.io

File: .github/workflows/java-microservice-ci-default.yml [hunk 7]
```diff
@@ -0,0 +1,679 @@
...
+        description: "The sonar token"
+        required: true
+      AWS_ACCESS_KEY_ID:
+        description: "The AWS access key"
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        description: "The AWS secret access key"
+        required: true
+
+jobs:
```
Comment (alexandre-cartapanis) [inline .github/workflows/java-microservice-ci-default.yml:82]: Maybe we can try to split in some more jobs.

Take a look at https://github.com/liquibase/liquibase/blob/master/.github/workflows/build.yml

I know artifacts sharing is not possible with docker images for us, but maybe we can find interesting things in this build.

File: .github/workflows/java-microservice-ci-default.yml [hunk 8]
```diff
@@ -0,0 +1,679 @@
...
+      chart_museum_url:
+        description: "The ChartMuseum URL"
+        type: string
+        required: true
+      enable_trivy_exit:
+        description: "Enable Trivy failure when CVE are found, you can deactivate this behavior"
+        type: boolean
+        required: false
+        default: true
+      enable_database_docker:
```
[the remark this answers is not in the document]
Comment (sebastienrospars) [reply .github/workflows/java-microservice-ci-default.yml:63]: Sur des reusable workflow c'est pas possible de faire des matrices car c'est tout le job qui est matrixé

File: .github/workflows/java-microservice-ci-default.yml [hunk 9]
```diff
@@ -0,0 +1,679 @@
...
+        description: "The project name"
+        type: string
+        required: true
+      aws_region:
+        description: "The AWS region"
+        type: string
+        required: false
+        default: "eu-west-1"
+      ecr_url:
+        description: "The ECR url"
```
[the remark this answers is not in the document]
Comment (sebastienrospars) [reply .github/workflows/java-microservice-ci-default.yml:51]: Le workflow est publique (car pas possible de les avoirs privé) donc je mets aucune info

File: .github/workflows/java-microservice-ci-default.yml [hunk 10]
```diff
@@ -0,0 +1,679 @@
...
+        type: string
+        required: true
+      chart_museum_url:
```
[the remark this answers is not in the document]
Comment (sebastienrospars) [reply .github/workflows/java-microservice-ci-default.yml:54]: Same as above

**Pull request merged**